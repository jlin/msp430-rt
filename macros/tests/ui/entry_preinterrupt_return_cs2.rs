#![no_main]

use msp430::interrupt::CriticalSection;
use msp430_rt_macros::entry;

fn init(cs: CriticalSection) -> CriticalSection {
    cs
}

#[entry(interrupt_enable(pre_interrupt = init))]
fn main(_cs: CriticalSection) -> ! {
    loop {}
}
