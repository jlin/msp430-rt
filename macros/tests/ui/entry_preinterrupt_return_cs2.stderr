error[E0716]: temporary value dropped while borrowed
  --> tests/ui/entry_preinterrupt_return_cs2.rs:10:1
   |
10 | #[entry(interrupt_enable(pre_interrupt = init))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
   | |                                              |
   | |                                              temporary value is freed at the end of this statement
   | creates a temporary value which is freed while still in use
   | borrow later stored here
   |
   = note: this error originates in the attribute macro `entry` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider using a `let` binding to create a longer lived value
   |
10 + let binding = #[entry(interrupt_enable(pre_interrupt = init))];
11 ~ binding
   |
